libcontainer/mount_linux.go

```diff
@@ -121,3 +179,75 @@ func syscallMode(i fs.FileMode) (o uint32) {
 	// No mapping for Go's ModeTemporary (plan9 only).
 	return
 }
+
+// mountFd creates an open_tree(2)-like mount fd from the provided
+// configuration. This function must be called from within the container's
+// mount namespace.
+func mountFd(nsHandles *userns.Handles, m *configs.Mount) (mountSource, error) {
+	if !m.IsBind() {
+		return mountSource{}, errors.New("new mount api: only bind-mounts are supported")
+	}
+	if nsHandles == nil {
+		nsHandles = new(userns.Handles)
+		defer nsHandles.Release()
+	}
+
+	var mountFile *os.File
+	var sourceType mountSourceType
+
+	if m.IsBind() {
+		// We only need to use OPEN_TREE_CLONE in the case where we need to use
+		// mount_setattr(2). We are currently in the container namespace and
+		// there is no risk of an opened directory being used to escape the
+		// container. OPEN_TREE_CLONE is more expensive than open(2) because it
+		// requires doing mounts inside a new anonymous mount namespace.
+		if m.IsIDMapped() {
+			flags := uint(unix.OPEN_TREE_CLONE | unix.O_CLOEXEC)
+			if m.Flags&unix.MS_REC == unix.MS_REC {
+				flags |= unix.AT_RECURSIVE
+			}
+			mountFd, err := unix.OpenTree(unix.AT_FDCWD, m.Source, flags)
```

Comment:
How about rename the var name `mountFd` to `Fd`? Because its name is the same as the function name.